semgrep/semgrep/target_manager.py

```diff
@@ -54,8 +55,10 @@ def lang_to_exts(language: Language) -> List[FileExtension]:
     """
     if language in {"python", "python2", "python3", "py"}:
         return PYTHON_EXTENSIONS
-    elif language in {"js", "javascript"}:
+    elif language in {"js", "jsx", "javascript"}:
         return JAVASCRIPT_EXTENSIONS
+    elif language in {"ts", "tsx" "typescript"}:
```

Comment:
Do we need to filter out `.d.ts`?

Reply:
A hack I used for parsing stats is to treat `.d.ts` as the extension of `foo.d.ts`. I think it violates the usual meaning of "extension". Applying a more generic filter on file names seems like a proper solution here.